docs/basic_usage.md

```diff
@@ -0,0 +1,62 @@
+# DAPper Basic Usage Guide
+
+DAPper (Dependency Analysis Project) is a tool that analyzes C/C++ source code to discover implicit dependencies and identify which system packages are required. This guide covers the basic usage patterns and common workflows.
+
+## Overview
+
+DAPper helps you:
+- **Discover header dependencies**: Find all `#include` statements in your C/C++ code
+- **Identify system packages**: Map header files to the packages that provide them
+- **Analyze large codebases**: Process entire directories recursively
+- **Understand implicit dependencies**: Reveal dependencies that aren't immediately obvious
+
+## Command Line Interface
+
+### Basic Syntax
+
+```bash
+# Analyze a single file
+dapper <source_file>
+
+# Analyze a directory
+dapper <source_directory>
+
+# Using cargo 
+cargo run -- <source file or directory>
+```
+
+## Understanding the Output
+
+### Basic Output Format
+
+DAPper outputs the `#include` files found in each C/C++ source file. Here's what to expect:
+
+**Example input file (`test.cpp`):**
+```cpp
+#include <iostream>
+#include "test.h"
+
+using namespace std; 
+
+int main(int argc, char* argv[])
+{
+    cout << "Hello World!" << endl;
+    return 0;
+}
+```
+
+**Example DAPper output:**
+```
+CPP(UserInclude("test.h")):
+        ["libcgal-dev", "asterisk-dev", "libboost1.67-dev", "libcxxtools-dev", "libwolfssl-dev", ...]
+```
+
+### Output Interpretation
+
+- Components:
+    - **CPP**: Indicates this is a C++ source file being analyzed
+    - **UserInclude("test.h")**: Shows DAPper found a local/user include directive #include "test.h"
```

Comment:
This part could then also mention the SystemInclude. (Basically something a bit more unique than "test.h" which seems to appear in many packages)